details/encryption.md

```diff
@@ -0,0 +1,382 @@
+
+The user agent then determines which helper party network will receive the
+information.  Superficially, this requires that the user agent query an HTTPS
+resource at the site origin to retrieve the identity of the helper party
+network, but as the protocol requires that sites
+[commit](https://github.com/patcg-individual-drafts/ipa/blob/main/IPA-End-to-End.md#commitments)
+to a helper party network, this might involve a different process.  The specific
+details of this process are not yet determined.
+
+The user agent then obtains:
+
+* The identity of the three helper parties in the site's chosen helper party
+  network.
+
+* The key configuration for each helper party.
+
+### Replicated Secret Sharing
+
+The user agent then creates three binary sharings ($mk_1$, $mk_2$, $mk_3$) of
+the match key by generating two random values (see [RFC
+4086](https://datatracker.ietf.org/doc/html/rfc4086)), setting the first two
+shares to these values, then setting the third to the exclusive OR of the match
+key and the two random values.
+
+```python
+mk[1] = random()
+mk[2] = random()
+mk[3] = mk[1] ^ mk[2] ^ match_key
+```
+
+The user agent then allocates two of the three shares to each helper, such that
+each share is sent to two different helpers.  For example, $P_1$ receives $mk_1$
+and $mk_2$, $P_2$ receives $mk_2$ and $mk_3$, and $P_3$ receives $mk_3$ and
+$mk_1$.  This is a particular form of replicated secret sharing that lends
+itself to efficient computation in an honest majority setting.
+
+<aside>
+
+A note on notation here.  In a few places, it is necessary to describe helpers
+or shares relative to a current helper.  In this case, we will use $P_i$ to
+refer to the current helper and $mk_i$ to refer to the corresponding share.
+Helpers are formed into a ring, with higher-numbered helpers to the right of the
+helper immediately preceding them.  Thus, the helper to the right of the current
+helper is identified as $P_{i+1}$.  As this is a ring, the helper to the right
+of $P_3$ is not $P_4$, but $P_1$.  Similarly, the previous helper is identified
+as $P_{i-1}$ and the helper to the left of $P_1$ is $P_3$.
+
+In this view, the replicated secret sharing we use has each helper with a share
+in common with the helper to its left ($mk_i$) and a share in common with the
+helper to its right ($mk_{i+1}$).  While it would be equally valid (and
+indistinguishable from this) if we instead used $mk_{i-1}$ and $mk_i$, we use
+$mk_i$ and $mk_{i+1}$ throughout.
+
+</aside>
+
+
+### HPKE Context Creation
+
+The two values destined for each helper are then concatenated and encrypted
+toward that helper using HPKE.  The sender and receiver contexts used by HPKE
+are constructed using an input that captures critical contextual information.
+
+The `info` input for HPKE shall include a distinguishing sequence that is
+composed from:
+
+1. The fixed string "private-attribution", encoded in ASCII,
+
+2. The ASCII serialization of the helper party origin.
```

Comment:
The system doesn't care who makes the query, does it?  Trigger sites pay in terms of privacy budget for trigger queries, not the report collector.

The reason we need (well, not really, it's a bit of defense in depth) this is so that the browser can't be duped into directing values to a different helper party.

The origin of the record collector doesn't really matter if they can all spend the privacy budget equally.  Though if we're talking about apportioning budget across different different record collectors, then maybe it makes sense to include something here.  But I don't think that the browser needs to be involved in that.  Sites can work that out between themselves, the helper parties, and any report collectors.